src/linux/process.rs

```diff
@@ -219,6 +219,25 @@ impl ProcessExt for Process {
     fn group_id(&self) -> Option<Gid> {
         self.group_id
     }
+
+    fn wait(&self) -> Option<Pid> {
+        let (mut status, mut _pid): (libc::c_int, i32) = (0, -1);
+        let _ppid = self.parent().unwrap().as_u32();
+        let _uppid = std::process::id();
+        if _ppid == _uppid {
+            unsafe {
+                Some(Pid(libc::waitpid(self.pid.0, &mut status, 0)))
+            }
+        } else {
+            while self.kill_with(Signal::Null).unwrap_or(false) {
+                unsafe {
+                    Pid(libc::waitpid(self.pid.as_u32() as i32, &mut status, 0))
```

Comment:
Like I said in the issue, `waitpid` only works for children processes, not for any processes. This is basically doing nothing here.

Reply:
From `waitpid` manual:

>  All  of  these  system  calls  are  used  to  wait for state changes in a child of the calling process, and obtain information about the child whose state has changed.

The only thing we can do is looping on the `kill` call.